### PR TITLE
Migrated to beta01 and added compose navigation

### PR DESCRIPTION
**Pull request by sante**
## Description:

- Migrated to compose beta01
- Added compose navigation

## :pushpin: Notes:

- The app will need a refactor before the epic is merged to improve how the screens are made
- ~Because of some compose changes, had to provide `ViewModels` with their provider instead of Koin, but inside the `ViewModels` I still used Koin~

### CODE REVIEW

File: app/src/main/java/com/xmartlabs/gong/ui/screens/welcome/WelcomeScreen.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      Column(
+          verticalArrangement = Arrangement.Center,
+          modifier = Modifier.fillMaxSize()
+      ) {
+        val userResult by viewModel.userLiveData.observeAsState()
+        Text(
+            text = "Hi ${userResult?.getOrNull()?.name}",
+            style = MaterialTheme.typography.h3.copy(
+                fontSize = 40.sp,
+                color = colorResource(id = android.R.color.darker_gray)
```
Comment (f7deleon) [inline app/src/main/java/com/xmartlabs/gong/ui/screens/welcome/WelcomeScreen.kt:45]: shoudn't colors be obtain from the material theme?

**Review by f7deleon — APPROVED** [review]
Nice!! :tada:

File: app/src/main/java/com/xmartlabs/gong/ui/screens/welcome/WelcomeScreen.kt (hunk 1 above)
Comment (sante) [reply app/src/main/java/com/xmartlabs/gong/ui/screens/welcome/WelcomeScreen.kt:45]: Ideally, we cwould have a custom theme defined and pull the colors from there, waiting on the new design to do that

File: app/build.gradle [hunk 2]
```diff
@@ -3,7 +3,7 @@ buildscript {
     dependencies {
         classpath buildscriptDeps.firebaseCrashlyticsPlugin
         classpath buildscriptDeps.snapshotPublisher
-        classpath buildscriptDeps.navigationSafeArgPlugin
+        //classpath buildscriptDeps.navigationSafeArgPlugin TODO enable when compatible
```
Comment (mirland) [inline app/build.gradle:6]: but, does it need? I think it's not needed anymore, right?

File: app/src/main/java/com/xmartlabs/gong/ui/MainActivity.kt [hunk 3]
```diff
@@ -1,13 +1,36 @@
...
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
 import com.xmartlabs.gong.R
+import com.xmartlabs.gong.ui.screens.signin.SignInScreen
+import com.xmartlabs.gong.ui.screens.splash.SplashScreen
+import com.xmartlabs.gong.ui.screens.welcome.WelcomeScreen
+import kotlin.time.ExperimentalTime
 
+@ExperimentalTime
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/gong/ui/MainActivity.kt:17]: add the compilation flags

File: app/src/main/java/com/xmartlabs/gong/ui/screens/signin/SignInScreen.kt [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    onPasswordEdited: (String) -> Unit = {},
+    onSignInButtonClicked: () -> Unit = {},
+) {
+  Surface(color = Color.White) {
+    ConstraintLayout(modifier = Modifier
+        .fillMaxWidth()
+        .fillMaxHeight()) {
+      val (hiAllTextView, userIdEditText, passwordEditText, signInButton, viewModelTimeTextView) = createRefs()
+
+      Text(text = "Hi all", fontSize = 20.sp, modifier = Modifier.constrainAs(hiAllTextView) {
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/gong/ui/screens/signin/SignInScreen.kt:88]: ml?

File: app/src/main/java/com/xmartlabs/gong/ui/screens/signin/SignInScreen.kt [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+  }
+}
+
+private fun signIn(
+    viewModel: SignInScreenViewModel,
+    lifecycleOwner: LifecycleOwner,
+    context: Context,
+    navController: NavHostController,
+) =
+    with(viewModel) {
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/gong/ui/screens/signin/SignInScreen.kt:154]: same line?

**Review by mirland — APPROVED** [review]

**Review by mirland — APPROVED** [review]

**Pull request merged**